### ISSUE TITLE
Display a list of goods that are desired to be sold and eventually sold in Trading/Delivery mission

### ISSUE THREAD

**Issue by mokun** [mars-sim/mars-sim#732]
**Describe the bug**
- Mission Tool only displays a list of desired goods to buy and eventually bought in Trading/Delivery mission
- It does not display a list of desired goods to sell and eventually being sold

**Expected and unexpected behavior**
- If Settlement A is selling good X to Settlement B in a trading/delivery mission, make sure Settlement A will not be buying back the very same good from Settlement B
- Display a list of desired goods to sell and eventually being sold

**Screenshots**
![image](https://user-images.githubusercontent.com/1168584/197103356-60c784a5-90b3-43ab-b8db-7b26f6128079.png)

**Specifications  (please complete the following information):**
 - pre-3.4.0 build 7582

**Comment by mokun** [comment]
@bevans2000 

Btw, to further simplify the display, we only need one table with 3 columns to show what good (col 1) is expected to be sold at what quantity (col 2) and actually being sold at what quantity (col 3).

**Comment by mokun** [comment]
The Mission Tool fails to show what settlement Schiaparelli Point is interested in trading with. 

![image](https://user-images.githubusercontent.com/1168584/197104861-3b77a9a8-c9fb-4f61-85b3-d75987ad3e99.png)

**Comment by mokun** [comment]
I'll have to use Monitor Tool to discover that which settlement it's trading with

![image](https://user-images.githubusercontent.com/1168584/197105062-6c4ff621-ae9e-47b6-8fff-e33699193907.png)

**Comment by mokun** [comment]
If I go to `New Boston` in Mission Tool, it doesn't show it has Trading 17. I suppose it's because Trading 17 is NOT started by by `New Boston` and therefore it won't show it there.

![image](https://user-images.githubusercontent.com/1168584/197105331-275f39fe-e3e4-46cb-8eb8-38f2ee98917d.png)

**Comment by bevans2000** [comment]
Lots of things in one issue

Yes, one table showing buy & sell would work. The fact a mission can buy & sell the same Goods is a totally separate issue and relates to how we score items. This will be how the ```GoodsManager``` creates ```Deals```; these are created periodically. The logic needs a tweak to exclude any Goods in the Sell Deal from the target Buy Deal. Should be straight forward as this logic is now encapsulated in the ```Deal``` class so it's a case of subtracting one from the other. Create a new issue for this problem. 

The Trading settlement should be added to the Trading panel; the top panel should be left alone as it is consistent for all Missions and I like that. This leads to the Mission not being shown for the Trading Settlement but only the Starting Settlement. I feel this is correct because the Mission tool selects Missions that are "owned" by a Settlement.

**Comment by mokun** [comment]
> The fact a mission can buy & sell the same Goods is a totally separate issue and relates to how we score items.

I was going to say when I first created this ticket I was assuming this could happen but later can't find evidence to confirm that.

So it may or may not happen.

Can you see any issue via console ?

**Comment by mokun** [comment]
In terms of the second problem, are we originally dealing with 4 lists (or 4 tables) ?

As you know, currently, the UI only show what items that settlement A initially suggest to sell and what items actually being sold. 

I suppose it won't show the other side of the story, namely, when the rover or delivery drone returns home, what items that settlement B would like to offer to sell and what items actually being accepted to be sold back. 

One reason I can't think of is that it can't be shown in Mission Tool because this *second* portion of the deal has NOT happened yet, at least not until the vehicle first arrived at settlement B and face-to-face negotiation needs to be made and come through regarding the *first* deal. 

After the *first* deal is settled, then the *second* deal begins on the spot. Is that true ?

**Comment by bevans2000** [comment]
> > The fact a mission can buy & sell the same Goods is a totally separate issue and relates to how we score items.
> 
> I was going to say when I first created this ticket I was assuming this could happen but later can't find evidence to confirm that.
> 
> So it may or may not happen.
> 
> Can you see any issue via console ?

The Console has a 'trade' command that shows the deals that have been calculated. Essentially in the new improved world, each Settlement maintains 2 ```Deal``` objects, one holding items it wants go sell and the other it is looking to buy. When it looks for Delivery mission it simple compares the deals options to make sure the profit cover the travel cost.
What  we need to change to fix this is to make sure any items in the Sell deal can not be added to the Buy Deal. Create a separate issue

**Comment by bevans2000** [comment]
> In terms of the second problem, are we originally dealing with 4 lists (or 4 tables) ?
> 
> As you know, currently, the UI only show what items that settlement A initially suggest to sell and what items actually being sold.
> 
> I suppose it won't show the other side of the story, namely, when the rover or delivery drone returns home, what items that settlement B would like to offer to sell and what items actually being accepted to be sold back.
> 
> One reason I can think of why it's not readily available in Mission Tool because this _second_ portion of the deal has NOT happened yet, at least not until the vehicle first arrived at settlement B and face-to-face negotiation needs to be made and come through regarding the _first_ deal.
> 
> It is after the _first_ deal is settled that the _second_ deal begins on the spot at settlement B.
> 
> Is that true ?

Correct, only the outbound trade is known. There is a desired deal  but it is not fixed until the mission arrives at the Trading settlement.

**Comment by mokun** [comment]
> Correct, only the outbound trade is known. There is a desired deal but it is not fixed until the mission arrives at the Trading settlement.

I suppose it has to be done this way since there's no such thing as currency on mars-sim and what settlement A and B are doing is almost like a form of bartering.

One obvious question is, since it's about *exchanging goods* between two settlements, why not consider what to get and what to receive at the same time between them at the beginning ?

**Comment by mokun** [comment]
@bevans2000 

Can we get players to involve more in settling a transaction prior to an agreement of exchanging good ?

I know realistically, there are *tradeoff* in every trade. One always have to give in something to get something.

Can we break down further on the decision matrix ? 

Can we give the players checkboxes and options to customize a trade/delivery ?

**Comment by mokun** [comment]
@bevans2000 

You might have been working on this task already on a different thread, have you ?
